src/main/java/com/esri/samples/imagelayers/query_map_image_sublayer/QueryMapImageSublayerSample.java

```diff
@@ -0,0 +1,225 @@
+      mapView.setMap(map);
+
+      // create a graphics overlay to show the results in
+      GraphicsOverlay graphicsOverlay = new GraphicsOverlay();
+      mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+      // create symbols for showing the results of each sublayer
+      SimpleMarkerSymbol citySymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, 0xFFFF0000, 16);
+      SimpleLineSymbol stateSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFF0000FF, 6);
+      SimpleLineSymbol countyLineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DASH, 0xFF00FFFF, 2);
+      SimpleFillSymbol countySymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.DIAGONAL_CROSS, 0xFF00FFFF,
+          countyLineSymbol);
+
+      // create a vbox to hold the controls
+      VBox vBoxControl = new VBox(6);
+      vBoxControl.setMaxSize(160, 100);
+      vBoxControl.getStyleClass().add("panel-region");
+
+      // create a spinner to input the population filter
+      Label label = new Label("Population greater than");
+      Spinner<Integer> populationSpinner = new Spinner<>(0, 100000000, 1800000);
+      populationSpinner.setEditable(true);
+
+      // create a button to execute the query, disable initially
+      Button queryButton = new Button("Query");
+      queryButton.setDisable(true);
+
+      // add the controls to the vbox
+      vBoxControl.getChildren().addAll(label, populationSpinner, queryButton);
+
+      // wait until the layer is loaded before enabling the query button
+      imageLayer.addDoneLoadingListener(() -> {
+        if (imageLayer.getLoadStatus() == LoadStatus.LOADED) {
+          queryButton.setDisable(false);
+
+          // get and load each sublayer to query
+          ArcGISMapImageSublayer citiesSublayer = (ArcGISMapImageSublayer) imageLayer.getSublayers().get(0);
+          ArcGISMapImageSublayer statesSublayer = (ArcGISMapImageSublayer) imageLayer.getSublayers().get(2);
+          ArcGISMapImageSublayer countiesSublayer = (ArcGISMapImageSublayer) imageLayer.getSublayers().get(3);
+          citiesSublayer.loadAsync();
+          statesSublayer.loadAsync();
+          countiesSublayer.loadAsync();
+
+          // query the sublayers when the button is clicked
+          queryButton.setOnAction(e -> {
+
+            // clear any previous results
+            graphicsOverlay.getGraphics().clear();
+
+            // create query parameters filtering based on population and the map view's current viewpoint
+            QueryParameters populationQuery = new QueryParameters();
+            populationQuery.setWhereClause("POP2000 > " + populationSpinner.getValue());
+            populationQuery.setGeometry(mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY)
+                .getTargetGeometry());
+
+            // query each sublayer's feature table with the query parameters and display the result features as graphics
+            citiesSublayer.addDoneLoadingListener(() -> {
```

Comment:
Shouldn't need to add done loading listener. I would check that it is loaded though.

Would do this for the other two as well.